src/webgpu/api/validation/queue/copyToTexture/CopyExternalImageToTexture.spec.ts

```diff
@@ -816,12 +816,14 @@ g.test('OOB,source')
   .paramsSubcasesOnly(u =>
     u
       .combine('srcOrigin', [
-        { x: 0, y: 0 }, // origin is on top-left
-        { x: kDefaultWidth - 1, y: 0 }, // x near the border
-        { x: 0, y: kDefaultHeight - 1 }, // y is near the border
-        { x: kDefaultWidth, y: kDefaultHeight }, // origin is on bottom-right
-        { x: kDefaultWidth + 1, y: 0 }, // x is too large
-        { x: 0, y: kDefaultHeight + 1 }, // y is too large
+        { x: 0, y: 0, z: 0 }, // origin is on top-left
+        { x: kDefaultWidth - 1, y: 0, z: 0 }, // x near the border
+        { x: 0, y: kDefaultHeight - 1, z: 0 }, // y is near the border
+        { x: kDefaultWidth, y: kDefaultHeight, z: 0 }, // origin is on bottom-right
+        { x: 0, y: 0, z: kDefaultDepth },
+        { x: kDefaultWidth + 1, y: 0, z: 0 }, // x is too large
+        { x: 0, y: kDefaultHeight + 1, z: 0 }, // y is too large
+        { x: 0, y: 0, z: kDefaultDepth + 1 }, // z is too large
       ])
       .expand('copySize', generateCopySizeForSrcOOB)
```

Comment:
I think `generateCopySizeForSrcOOB` needs an update to make this work because right now it always generates a copy size with `depthOrArrayLayers: 1`, but in the `{ x: 0, y: 0, z: kDefaultDepth }` case it should generate `depthOrArrayLayers: 0`.

Just need to extend it to do all the same things for z that it current does for x and y.

Reply:
Thank you for pointing it out. I update `generateCopySizeForSrcOOB` refer to `generateCopySizeForDstOOB` function.